handsontable/src/plugins/columnSummary/columnSummary.js

```diff
@@ -193,11 +194,15 @@ export class ColumnSummary extends BasePlugin {
     let biggestDecimalPlacesCount = 0;
 
     do {
-      cellValue = this.getCellValue(i, col) ?? 0;
-      const decimalPlaces = (((`${cellValue}`).split('.')[1] || []).length) || 1;
+      cellValue = this.getCellValue(i, col);
+      cellValue = isNullishOrNaN(cellValue) ? null : cellValue;
 
-      if (decimalPlaces > biggestDecimalPlacesCount) {
-        biggestDecimalPlacesCount = decimalPlaces;
+      if (cellValue !== null) {
+        const decimalPlaces = (((`${cellValue}`).split('.')[1] || []).length) || 1;
```

Comment:
This part of code looks horrible, but I know that it was just moved by you, so I don't focus on it.